bazel/envoy_build_system.bzl

```diff
@@ -0,0 +1,39 @@
+# References to Envoy external dependencies should be wrapped with this function.
+def envoy_external_dep_path(dep):
+    return "//external:%s" % dep
```

Comment:
Originally, Bazel didn't have a concept of named workspaces, where `workspace(name = ...)` can enforce how the workspace is imported into other workspaces (this is of course limited to repositories that support Bazel build system).

Because of that, people imported the same dependency under different names, for example BoringSSL could be imported as `git_new_repository(name = ssl)` or `git_new_repository(name = boringssl)` or `git_new_repository(name = boringssl_git)` and `//external` was a way to cheat around that with aliasing `@....//:ssl` to `//external:ssl`.

But since BoringSSL can enforce that it's always imported with `git_new_repository(name = boringssl)` and available as `@boringssl//`, there is no real reason to provide `//external:ssl` alias for `@boringssl//:ssl`, since the `:ssl` target is always going to be available under this path.